### PR TITLE
add log4j 1.2.17 metadata

### PR DESCRIPTION
**Pull request by FoghostCn**
## What does this PR do?

add metadata for log4j 1.2.17

## Checklist before merging
- [x] I have considered including reachability metadata directly in the library or the framework (see [our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md))
- [x] I am the original author of all content provided in the pull request, and I did not copy the content from any other source (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))
- [x] For all tests where I am not the sole author, I have added a comment that proves I may publish them under the specified license (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#tests))
- [x] I have properly formatted metadata files (see [our formatting guide](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#format-metadata-files))
- [x] I have added thorough tests (see [the tests section from our contributing document](https://github.com/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#Tests))
- [x] I have filled all places where my pull request accesses files, network, docker, or any other external service (see the section above)

### CODE REVIEW

File: tests/src/log4j/log4j/1.2.17/src/test/resources/log4j.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE log4j:configuration PUBLIC "-//log4j/log4j Configuration//EN" "log4j.dtd">
```
Comment (msupic) [inline tests/src/log4j/log4j/1.2.17/src/test/resources/log4j.xml:2]: Is this file copied from some web page?

**Review by msupic — CHANGES_REQUESTED** [review]

File: tests/src/log4j/log4j/1.2.17/src/test/resources/log4j.xml (hunk 1 above)
Comment (FoghostCn) [reply tests/src/log4j/log4j/1.2.17/src/test/resources/log4j.xml:2]: sorry for that, I'll fix it

File: tests/src/log4j/log4j/1.2.17/src/test/java/log4j/log4j/Log4jTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    private final PrintStream systemOut = System.out;
+
+    @BeforeEach
+    public void setUp() {
+        stdOut = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(stdOut));
+    }
+
+    @Test
+    void consoleInfoLogger() {
```
Comment (msupic) [inline tests/src/log4j/log4j/1.2.17/src/test/java/log4j/log4j/Log4jTest.java:34]: A few more things: The `consoleInfoLogger` and `consoleInfoLoggerProperties` tests create the `logs` folder and the following files in it:
```
dailyRollingAppender.log
fileAppender.log
rollingFileAppender.log
```
Can you check at the end of each test that the expected message exist in those files, something like:
```
    String loggedMessage = Files.readAllLines(Paths.get(filePath)).get(0);
    assertThat(loggedMessage).contains("info message ");
```
and delete the `logs` folder after each executed test. The `deletion` code can be added to the following method
```
    @AfterEach
    public void tearDown() {
        System.setOut(systemOut);
    }
```

**Review by msupic — CHANGES_REQUESTED** [review]

**Comment by msupic** [comment]
@FoghostCn - tests are failing. It looks like that some metadata are missing:
```
JUnit Jupiter:Log4jTest:consoleInfoLoggerProperties()
    MethodSource [className = 'log4j.log4j.Log4jTest', methodName = 'consoleInfoLoggerProperties', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class org.apache.log4j.LogManager
       org.apache.log4j.PropertyConfigurator.configure(PropertyConfigurator.java:415)
       log4j.log4j.Log4jTest.consoleInfoLoggerProperties(Log4jTest.java:52)
```

**Comment by FoghostCn** [comment]
I have regenerate a reflect-config.json file,it should be right

**Comment by msupic** [comment]
@FoghostCn - Tests are still failing. You can reproduce the issue locally by executing
`./gradlew test -Pcoordinates=log4j:log4j:1.2.17`

**Comment by FoghostCn** [comment]
Thank you so much,I've run successed locally

**Comment by msupic** [comment]
Resource metadata for `log4j.xml` and `log4j.dtd` already exists here: [metadata/log4j/log4j/1.2.17/resource-config.json](https://github.com/FoghostCn/graalvm-reachability-metadata/blob/master/metadata/log4j/log4j/1.2.17/resource-config.json), so you can just add the following entry to the same file instead of creating new resource-config.json file that is used only in tests
```
      {
        "condition": {
          "typeReachable": "org.apache.log4j.helpers.Loader"
        },
        "pattern": "\\Qlog4j.properties\\E"
      }
```

**Comment by FoghostCn** [comment]
OK,I will fix it tomorrow,I'm not next to the computer right now.

**Comment by FoghostCn** [comment]
> Resource metadata for `log4j.xml` and `log4j.dtd` already exists here: [metadata/log4j/log4j/1.2.17/resource-config.json](https://github.com/FoghostCn/graalvm-reachability-metadata/blob/master/metadata/log4j/log4j/1.2.17/resource-config.json), so you can just add the following entry to the same file instead of creating new resource-config.json file that is used only in tests
> 
> ```
>       {
>         "condition": {
>           "typeReachable": "org.apache.log4j.helpers.Loader"
>         },
>         "pattern": "\\Qlog4j.properties\\E"
>       }
> ```

done

**Review by msupic — APPROVED** [review]

**Pull request merged**